tests/test_simple_endpoint.py

```diff
@@ -29,3 +29,65 @@ def test_should_pass_validation(self):
         _, response = self._app.test_client.get('/', json={'name': 'john'})
         self.assertEqual(response.status, 200)
         self.assertEqual(response.json['status'], 'ok')
+
+
+class TestSimpleEndpointArgsValidation(unittest.TestCase):
+    _endpoint_schema = {'name': {'type': 'string', 'required': True}}
+    _app = None
+
+    def setUp(self):
+        self._app = Sanic()
+
+        @self._app.route('/')
+        @validate_args(self._endpoint_schema)
+        async def _simple_endpoint(request):
+            return json({'status': 'ok'})
+
+    def test_should_fail_for_empty_validation(self):
+        _, response = self._app.test_client.get('/')
+        self.assertEqual(response.status, 400)
+        self.assertEqual(response.json['error']['message'],
+                         'Validation failed.')
+        self.assertEqual(response.json['error']['type'], 'validation_failed')
+
+    def test_should_pass_validation(self):
+        _, response = self._app.test_client.get('/', params={'name': 'john'})
+        self.assertEqual(response.status, 200)
+        self.assertEqual(response.json['status'], 'ok')
+
+
+class TestSimpleEndpointArgsTypeNormalizationValidation(unittest.TestCase):
+    _endpoint_schema = {
+        'val': {'type': 'integer', 'required': True, 'coerce': int}}
+    _app = None
+
+    def setUp(self):
+        self._app = Sanic()
+
+        @self._app.route('/')
+        @validate_args(self._endpoint_schema)
+        async def _simple_endpoint(request):
+            return json({'status': 'ok'})
+
+    def test_should_fail_for_empty_validation(self):
+        _, response = self._app.test_client.get('/')
+        self.assertEqual(response.status, 400)
+        self.assertEqual(response.json['error']['message'],
+                         'Validation failed.')
+        self.assertEqual(response.json['error']['type'], 'validation_failed')
+
+    def test_should_pass_with_string_validation(self):
+        _, response = self._app.test_client.get('/', params={'val': '420'})
+        self.assertEqual(response.status, 200)
+        self.assertEqual(response.json['status'], 'ok')
+
+    def test_should_pass_with_int_validation(self):
+        _, response = self._app.test_client.get('/', params={'val': '420'})
```

Comment:
`{'val': '420'}`
A copy-pasting error? From the method name I'd expect it to be `{'val': 420}`.

This might be a good place to use a [parametrized test](https://docs.pytest.org/en/3.6.1/parametrize.html).

Reply:
Yupp :p That was most likely a copy paste.
I don't really think parametrized tests are needed and I don't know how to get the parametrization to work when we are testing the error message on failure as well.